src/lib/krb5/rcache/rc_base.c

```diff
@@ -177,34 +56,107 @@ krb5_rc_default(krb5_context context, krb5_rcache *idptr)
                            &profstr) == 0 && profstr != NULL) {
         ret = k5_expand_path_tokens(context, profstr, &rcname);
         profile_release_string(profstr);
-        ret = krb5_rc_resolve_full(context, idptr, rcname);
+        ret = k5_rc_resolve(context, rcname, rc_out);
         free(rcname);
         return ret;
     }
 
     /* Resolve the default type with no residual. */
-    return resolve_type_and_residual(context, "dfl", "", idptr);
+    return k5_rc_resolve(context, "dfl:", rc_out);
 }
 
 
 krb5_error_code
-krb5_rc_resolve_full(krb5_context context, krb5_rcache *idptr,
-                     const char *string_name)
+k5_rc_resolve(krb5_context context, const char *name, krb5_rcache *rc_out)
 {
     krb5_error_code ret;
-    char *type, *sep;
+    struct typelist *t;
+    const char *sep;
+    size_t len;
+    krb5_rcache rc = NULL;
 
-    *idptr = NULL;
+    *rc_out = NULL;
 
-    sep = strchr(string_name, ':');
+    sep = strchr(name, ':');
     if (sep == NULL)
         return KRB5_RC_PARSE;
+    len = sep - name;
 
-    type = k5memdup0(string_name, sep - string_name, &ret);
-    if (type == NULL)
-        return ret;
+    for (t = typehead; t != NULL; t = t->next) {
+        if (strncmp(t->ops->type, name, len) == 0 && t->ops->type[len] == '\0')
+            break;
+    }
+    if (t == NULL)
+        return KRB5_RC_TYPE_NOTFOUND;
 
-    ret = resolve_type_and_residual(context, type, sep + 1, idptr);
-    free(type);
+    rc = k5alloc(sizeof(*rc), &ret);
```

Comment:
Though they're both shims for it anyway, this should probably be `k5calloc()` for clarity.

Reply:
Clarity that the memory is zeroed, you mean?  k5alloc() was first added without a k5calloc() to go with it, and has always allocated zeroed memory (its documenting comment says so).  k5calloc() was later added to avoid having to worry about multiplication overflow.

Reply:
You're right - I misremembered and thought you preferred k5calloc for when zero-initialization was important.  Nevermind!